src/dlgProfilePreferences.cpp

```diff
@@ -1780,14 +1783,31 @@ void dlgProfilePreferences::slot_setLogDir()
     QString currentLogDir = QFileDialog::getExistingDirectory(
             this, tr("Where should Mudlet save log files?"), (mLogDirPath.isEmpty() ? lineEdit_logFileFolder->placeholderText() : mLogDirPath), QFileDialog::DontUseNativeDialog);
 
-    // Set the text of lineEdit_logFileFolder to the selected directory
     if (!currentLogDir.isEmpty() && currentLogDir != NULL) {
-        mLogDirPath = currentLogDir;
-        lineEdit_logFileFolder->setText(mLogDirPath);
-        // Disable pushButton_resetLogDir if the directory is set to the default path
-        pushButton_resetLogDir->setEnabled(mLogDirPath != mudlet::getMudletPath(mudlet::profileReplayAndLogFilesPath, pHost->getName()));
+        // Disable pushButton_resetLogDir and clear
+        // lineEdit_logFileFolder if the directory is set to the
+        // default path
+        if (currentLogDir == mudlet::getMudletPath(mudlet::profileReplayAndLogFilesPath, pHost->getName())) {
+            // clear mLogDirPath, which sets the directory where logs are saved
+            // to Mudlet's default log path.
+            mLogDirPath.clear();
+            lineEdit_logFileFolder->clear();
+            // Set the cursor position to the end of the lineEdit's placeholder
+            // text.
+            lineEdit_logFileFolder->setCursorPosition(lineEdit_logFileFolder->placeholderText().length());
+            pushButton_resetLogDir->setEnabled(false);
+        } else {
+            // set mLogDirPath to the selected directory
+            mLogDirPath = currentLogDir;
+            // If the directory is anything other than the default log
+            // directory, set the text of lineEdit_logFileFolder to the selected
+            // directory.
+            lineEdit_logFileFolder->setText(mLogDirPath);
+            // Set the cursor position to the end of the text.
+            lineEdit_logFileFolder->setCursorPosition(mLogDirPath.length());
```

Comment:
It is more self contained *IMHO* to set the cursor to the end of the text it is working on, i.e.:

    lineEdit_logFileFolder->setCursorPosition(lineEdit_logFileFolder->text().length());